### PR TITLE
Allow to use host in the master-slave replication

### PR DESCRIPTION
**Pull request by git-hulk**

### CODE REVIEW

File: src/replication.cc [hunk 1]
```diff
@@ -201,22 +201,18 @@ void ReplicationThread::CallbacksStateMachine::EvCallback(bufferevent *bev,
...
-    // NOTE: Connection error will not appear here, network err will be reported
-    // in ConnEventCB. the error here is something fatal.
-    LOG(ERROR) << "[replication] Failed to start state machine, err: " << strerror(errno);
-  }
-
-  auto s = Util::SockSetTcpKeepalive(bufferevent_getfd(bev), 120);
+  Status s = Util::SockConnect(repl_->host_, repl_->port_, &cfd);
   if (!s.IsOK()) {
-    LOG(ERROR) << "[replication] Failed to set tcp-keepalive, err:" << s.Msg();
+    LOG(ERROR) << "[replication] Failed to connect the master, err:" << s.Msg();
```
Comment (ShooterIT) [inline src/replication.cc:211]: Just one log is enough?

Comment (git-hulk) [reply src/replication.cc:211]: no, should return here. it's missing before, fix it now.

Comment (shangxiaoxiong) [reply src/replication.cc:211]: If return here, stateMachine terminates and will not run again. It seems unreasonable.

**Comment by ShooterIT** [comment]
Just as we discussed, i record one card https://github.com/bitleak/kvrocks/projects/1#card-62672516, we should do that ASAP @bitleak/kvrocks-contributors

**Review by ShooterIT — APPROVED** [review]

**Review by karelrooted — APPROVED** [review]

**Pull request merged**